### PR TITLE
Handle preview-only packages in the link checker

### PR DESCRIPTION
**Pull request by scbedd**
If there is no GA release for the package, we will necessarily never have the GA named readme show up. (GA vs Preview readmes MUST be differently named to render properly on docs.ms)

This will clean up our checks. 

@weshaggard

### CODE REVIEW

**Comment by weshaggard** [comment]
We will have to do something similar in https://github.com/Azure/azure-sdk/blob/master/_includes/releases/pkgrow.md which is where the table gets displayed on the site. If you aren't familiar with Jekyll I can add this change.

**Comment by scbedd** [comment]
@weshaggard I may ping you, but I think I should be able to get it working on my fork.

**Comment by scbedd** [comment]
@weshaggard  I'm happy with how this renders, but the build now breaks because I introduced a bunch of failing links :) 

I'll use this new list of broken links to find all the unpublished readmes and get them resolved. I generated a whole bunch yesterday based off of _releases/latest, so I'm hoping it should be fairly quick.

File: _includes/releases/pkgrow.md [hunk 1]
```diff
@@ -22,11 +22,12 @@
 
     {% if item.VersionGA != "" %}
         {% assign url = msdocs_url | replace: 'item.Version', item.VersionGA %}
-        {% include releases/pkgbadge.md label="msdocs" url=url version=item.VersionGA %}
+        {% include releases/pkgbadge.md label="msdocs" url=url version=item.VersionGA | remove: 'item.PreSuffix' %}
```
Comment (weshaggard) [inline _includes/releases/pkgrow.md]: Why do the removal here? Can this not just all be added in the preview section? Just use append https://shopify.github.io/liquid/filters/append/.

**Comment by scbedd** [comment]
Before this can merge, cleanup updates from #1489 need to all merge through.

File: eng/scripts/Update-Release-Versions.ps1 [hunk 2]
```diff
@@ -34,13 +34,17 @@ function CheckLink($url)
...
+
 function UpdateDocLinks($lang, $pkg)
 {
   $version = $pkg.VersionGA
   if ($version -eq "") { $version = $pkg.VersionPreview }
 
   $trimmedPackage = $pkg.Package -replace "@?azure[\.\-/]", ""
-  $msdocvalid = CheckLink "https://docs.microsoft.com/${lang}/api/overview/azure/${trimmedPackage}-readme/"
+
+  if ($version -eq $pkg.VersionPreview) { $suffix = "-pre" }
```
Comment (weshaggard) [inline eng/scripts/Update-Release-Versions.ps1:45]: I see in the templates above you also include msdocs_preview_moniker_suffix is that not needed during this check?

**Comment by scbedd** [comment]
For failure on 7/9. The following is true:

previous release, need resolution

```
https://docs.microsoft.com/dotnet/api/overview/azure/Security.KeyVault.Certificates-readme-pre?view=azure-dotnet-preview
https://docs.microsoft.com/dotnet/api/overview/azure/Security.KeyVault.Keys-readme-pre?view=azure-dotnet-preview
https://docs.microsoft.com/dotnet/api/overview/azure/Security.KeyVault.Secrets-readme-pre?view=azure-dotnet-preview
```


Resolved in a `Smoke-Test` branch

```
https://docs.microsoft.com/dotnet/api/overview/azure/Messaging.EventHubs-readme-pre?view=azure-dotnet-preview
https://docs.microsoft.com/dotnet/api/overview/azure/Messaging.EventHubs.Processor-readme-pre?view=azure-dotnet-preview
https://docs.microsoft.com/dotnet/api/overview/azure/Storage.Blobs.Batch-readme-pre?view=azure-dotnet-preview
https://docs.microsoft.com/dotnet/api/overview/azure/Storage.Files.DataLake-readme-pre?view=azure-dotnet-preview
https://docs.microsoft.com/dotnet/api/overview/azure/Storage.Files.Shares-readme-pre?view=azure-dotnet-preview
https://docs.microsoft.com/java/api/overview/azure/storage-blob-batch-readme-pre?view=azure-java-preview
https://docs.microsoft.com/java/api/overview/azure/storage-blob-cryptography-readme-pre?view=azure-java-preview
https://docs.microsoft.com/java/api/overview/azure/storage-file-datalake-readme-pre?view=azure-java-preview
https://docs.microsoft.com/java/api/overview/azure/storage-file-share-readme-pre?view=azure-java-preview
https://docs.microsoft.com/java/api/overview/azure/storage-queue-readme-pre?view=azure-java-preview
https://docs.microsoft.com/javascript/api/overview/azure/event-hubs-readme-pre?view=azure-node-preview
https://docs.microsoft.com/javascript/api/overview/azure/storage-blob-readme-pre?view=azure-node-preview
https://docs.microsoft.com/javascript/api/overview/azure/storage-file-datalake-readme-pre?view=azure-node-preview
https://docs.microsoft.com/javascript/api/overview/azure/storage-file-share-readme-pre?view=azure-node-preview
https://docs.microsoft.com/javascript/api/overview/azure/storage-queue-readme-pre?view=azure-node-preview
https://docs.microsoft.com/python/api/overview/azure/storage-blob-readme-pre?view=azure-python-preview
https://docs.microsoft.com/python/api/overview/azure/storage-file-datalake-readme-pre?view=azure-python-preview
https://docs.microsoft.com/python/api/overview/azure/storage-file-share-readme-pre?view=azure-python-preview
```

Resolving the comment and seeing if there is a good way to address this.

File: eng/scripts/Update-Release-Versions.ps1 (hunk 2 above)
Comment (scbedd) [reply eng/scripts/Update-Release-Versions.ps1:45]: Currently this is not necessary, purely by virtue of how these documents are being copied over. Specifically, `latest` has ALL the readmes present, but you only need the moniker suffix if you want it to render the actual page on docs.ms.com properly. If we're just retrieving it's good enough.

File: _includes/releases/pkgrow.md (hunk 1 above)
Comment (scbedd) [reply _includes/releases/pkgrow.md]: Suggestion. Did so.

**Pull request merged**